agent/dns.go

```diff
@@ -575,6 +588,21 @@ func (d *DNSServer) formatNodeRecord(addr, qName string, qType uint16, ttl time.
 			}
 		}
 	}
+
+	if node != nil && len(node.Meta) > 0 && (qType == dns.TypeANY || qType == dns.TypeTXT) {
```

Comment:
Add test in case we are called from line [323](https://github.com/hashicorp/consul/pull/3343/commits/f498c53dc8b1e7aaf06d6bd291428bf07568a879#diff-0f2130d66e011e1af6e42f93492a3536R323).